users: bulk add users from csv file

FaiyazMujawar opened the pull request:
Fixes #50  

In this PR,
- Implemented functionality to add users from `.csv` file.
- Updated the documentation accordingly.

**PS:** I've used the [papaparse](https://www.npmjs.com/package/papaparse) npm package to parse the CSV file.

drigio commented on routes/user.js:
```diff
@@ -32,6 +34,14 @@ router.post(
 );
 
+router.post(
+  "/add/all",
```
should we use `bulk` instead of `all`?

FaiyazMujawar replied:
Okay.

drigio commented on services/FileService.js:
```diff
@@ -28,6 +28,14 @@ const saveImages = multer({
+ * Express middleware for storing csv containing user details on local
+ * storage.
+ */
+const saveCsv = multer({
```
What are we doing about the files once users have been created?

FaiyazMujawar replied:
It's being deleted. `unlinkSync() ` is being called. It's deleting the file.

drigio replied:
Okay. I guess i missed that part.

drigio commented on controllers/userController.js:
```diff
@@ -62,6 +64,69 @@ const addUser = async (req, res) => {
+    );
+
+    if (error) {
+      erroneousUsers.push(zprn);
```
I think we shouldn't add any user if there is an error with some user. Because if there are errors we should just return the `erroneous` users list and let the end user make changes in the csv regarding the same and then reupload the whole csv. Because some end users might reupload the same csv multiple times, creating more errors because of same zprn again repeating.

FaiyazMujawar replied:
Okay.

drigio replied:
Also lets make that clear in documentation and even in the response message that after there is an error, no user is added.

FaiyazMujawar committed:
users: add users from file

In this PR,
- Implemented functionality to add users from `.csv` file.
- Updated the documentation accordingly.

drigio reviewed (approved):
LGTM! Merging in
